idea/data-storage/src/meta/meta.service.ts

```diff
@@ -26,10 +26,28 @@ export class MetaService {
     private gearEventListener: GearEventListener,
   ) {}
 
+  public async getByHashApi(hash: string): Promise<Meta> {
```

Comment:
I'm sure that `getOrCreate` is musch better

Reply:
I don't think so. It's bad practice to use `or` when naming `function or method` in my opinion. But also not a good idea when function does more than one thing. For that reason before I'm not combining two actions in one method.

For that situation much better to use `get` but inside `method` we know that used two separate methods.

Reply:
Anyway getByHashApi means nothing

Reply:
Why nothing? I didn't get you. Do you think it's a good practice when `function` do more than one thing?

Reply:
I think that name of function should be meaningful and reflect what function does. What does HashApi mean?

Reply:
The method not called `HashApi` is correct name is `getByHashApi`. The `Api` prefix I have added for recognizing that method is used in `gearEventListerner`.

If some function will do more than one thing it will look like `getOrCreateOrUpdate`. May be meaningful but is not a good naming of a function and not a good idea to create like this function.

Reply:
In Rust, for example, you can defentely find many standart functions called `get_or_create` or `get_or_insert` etc. So I'm not agree with you that it's bad way. Of course more often it's not good to create functions that do a lot of actions but in this particulare case it looks justified.